ollama/_utils.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+from collections import defaultdict
+import inspect
+from typing import Callable, Union
+
+import pydantic
+from ollama._types import Tool
+
+
+def _parse_docstring(doc_string: Union[str, None]) -> dict[str, str]:
+  parsed_docstring = defaultdict(str)
+  if not doc_string:
+    return parsed_docstring
+
+  key = hash(doc_string)
+  for line in doc_string.splitlines():
+    lowered_line = line.lower()
```

Comment:
might be a good idea to `strip` here so it'll match trailing whitespace on `args:`

Reply:
Yes smart!